### PR TITLE
New Component: Add docs for Google Maps Location Sharing

### PR DESCRIPTION
**Pull request by michaelarnauts**
**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#12301

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Review by c727 — DISMISSED** [review]

File: source/_components/device_tracker.google_maps.markdown [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+footer: true
+logo: google_maps.png
+ha_release: 0.64
+ha_category: Presence Detection
+ha_iot_class: "Cloud Polling"
+---
+
+The `google_maps` platform allows you to detect presence using the unofficial API of [Google Maps Location Sharing](https://myaccount.google.com/locationsharing). 
+
+You first need to create an additional Google account, and share your location with that account. This platform will use that account to fetch the location of your device(s). You have to setup sharing trough the Google Maps app on your mobile phone. You can find more information [here](https://support.google.com/accounts?p=location_sharing).
```
Comment (PRabahy) [inline source/_components/device_tracker.google_maps.markdown]: Typo. "trough" is probably supposed to be "through".

File: source/_components/device_tracker.google_maps.markdown [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+```yaml
+# Example configuration.yaml entry
+device_tracker:
+  platform: google_maps
+  username: example@gmail.com
+  password: password
+```
+
+Configuration variables:
```
Comment (frenck) [inline source/_components/device_tracker.google_maps.markdown]: Please use the configuration tags, for more information please see:
https://home-assistant.io/developers/documentation/create_page/#configuration

**Review by frenck — CHANGES_REQUESTED** [review]
@michaelarnauts Thank you for this PR. I've reviewed it and left you a comment. Could you please take a look? 👍

**Comment by frenck** [comment]
Rebased the PR in order to fix the build

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Did some small updates myself.

This one can be merged in as soon as the build passes since the parent PR has been merged already.

File: source/_components/device_tracker.google_maps.markdown [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+Since this platform is using an official API with the help of [locationsharinglib](https://github.com/costastf/locationsharinglib), Google seems to block access to your data the first time you've logged in with this component.
+This issue can be fixed by logging in with your new account and approving your login on the [Device Activity](https://myaccount.google.com/device-activity) page.
+</p>
+
+To integrate Google Maps Location Sharing in Home Assistant, add the following section to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+device_tracker:
+  platform: google_maps
```
Comment (Skaronator) [inline source/_components/device_tracker.google_maps.markdown:32]: shouldn't this be:

```
device_tracker:
  - platform: google_maps
    username: example@gmail.com
    password: password
```
(missing `-`)?

**Pull request merged**